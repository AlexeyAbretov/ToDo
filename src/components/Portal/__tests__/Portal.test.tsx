/* eslint-disable import/no-extraneous-dependencies */
// eslint-disable import/no-extraneous-dependencies /
// eslint-disable no-undef /

import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@theme';
import { Portal } from '..';

const getUi = (node?: Element): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Portal domNode={node}>test</Portal>
    </ThemeProvider>
  );
};

describe('Portal tests', () => {
  afterEach(() => {
    document.body.innerHTML = '';
  });

  it('should render by body', () => {
    const customNode = document.createElement('div');
    customNode.setAttribute('id', 'customNode');
    document.body.appendChild(customNode);

    const { baseElement, unmount } = render(getUi());

    expect(baseElement).toMatchSnapshot();

    unmount();
  });

  it('should render in root', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'root');
    document.body.appendChild(root);

    const customNode = document.createElement('div');
    customNode.setAttribute('id', 'customNode');
    document.body.appendChild(customNode);

    const { baseElement, unmount } = render(getUi());

    expect(baseElement).toMatchSnapshot();

    unmount();
  });

  it('should render in custom node', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'root');
    document.body.appendChild(root);

    const customNode = document.createElement('div');
    customNode.setAttribute('id', 'customNode');
    document.body.appendChild(customNode);

    const { baseElement, unmount } = render(
      getUi(document.getElementById('customNode')),
    );

    expect(baseElement).toMatchSnapshot();

    unmount();
  });
});
