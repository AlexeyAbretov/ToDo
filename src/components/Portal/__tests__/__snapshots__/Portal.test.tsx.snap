// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal tests should render by body 1`] = `
<body>
  <div
    id="customNode"
  />
  <div />
  test
</body>
`;

exports[`Portal tests should render in custom node 1`] = `
<body>
  <div
    id="root"
  />
  <div
    id="customNode"
  >
    test
  </div>
  <div />
</body>
`;

exports[`Portal tests should render in root 1`] = `
<body>
  <div
    id="root"
  >
    test
  </div>
  <div
    id="customNode"
  />
  <div />
</body>
`;
